File: src/features/sequin/api/sequin.ts
import axios from 'axios';

const API_URL = 'http://localhost:8000/api/v1';

export const sequinApi = {
    listDatabases: async () => {
        const response = await axios.get(`${API_URL}/sequin/databases`);
        return response.data;
    },
    listSinks: async () => {
        const response = await axios.get(`${API_URL}/sequin/sinks`);
        return response.data;
    },
    createSink: async (data: any) => {
        const response = await axios.post(`${API_URL}/sequin/sinks`, data);
        return response.data;
    },
    createBackfill: async (sinkId: string, data: any) => {
        const response = await axios.post(`${API_URL}/sequin/sinks/${sinkId}/backfills`, data);
        return response.data;
    }
};

File: backend/app/api/v1/endpoints/sequin.py
from fastapi import APIRouter, Depends, HTTPException
from sqlmodel import Session
from app.core.database import get_session
from app.services.sequin import SequinService
from typing import Any, Dict

router = APIRouter()

@router.get("/databases")
async def list_databases(session: Session = Depends(get_session)):
    service = SequinService(session)
    return await service.list_databases()

@router.get("/sinks")
async def list_sinks(session: Session = Depends(get_session)):
    service = SequinService(session)
    return await service.list_sinks()

@router.post("/sinks")
async def create_sink(sink_data: Dict[str, Any], session: Session = Depends(get_session)):
    service = SequinService(session)
    return await service.create_sink(sink_data)

@router.post("/sinks/{sink_id}/backfills")
async def create_backfill(sink_id: str, backfill_data: Dict[str, Any], session: Session = Depends(get_session)):
    service = SequinService(session)
    return await service.create_backfill(sink_id, backfill_data)
